### PR TITLE
feat: non adjacent update

### PR DESCRIPTION
**Pull request by ibrizsabin**
### Commit Message

```bash
feat: non adjacent update
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
version: <log entry>
```

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the unit tests
- [ ] I only have one commit (if not, squash them into one commit).
- [ ] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

File: contracts/cosmwasm-vm/cw-icon-light-client/src/light_client.rs [hunk 1]
```diff
@@ -117,40 +126,55 @@ impl ILightClient for IconClient<'_> {
...
+            .get_consensus_state(client_id, signed_header.trusted_height)?;
+
+        let current_proof_context_hash =
+            btp_header.get_next_proof_context_hash(&signed_header.current_validators);
+
+        if current_proof_context_hash != trusted_consensus_state.next_proof_context_hash {
+            return Err(ContractError::InvalidProofContextHash);
+        }
+
+        if (btp_header.main_height - signed_header.trusted_height) > state.trusting_period {
```
Comment (AntonAndell) [inline contracts/cosmwasm-vm/cw-icon-light-client/src/light_client.rs:153]: We should also include a check:
assert(clientState.latestHeight - header.mainHeight < clientState.trustingPeriod)

File: docs/adr/ICON-lightclient.md [hunk 2]
```diff
@@ -233,18 +250,24 @@ Verify validity of regular update to the client
...
+
     // assert header has correct networkId
     assert(header.networkId == NETWORK_ID)
+
+    assert(header.trustedHeight < header.mainHeight)
     
     // assert trusting period has not yet passed
-    assert(header.mainHeight - clientState.latestHeight < clientState.trustingPeriod)
+    assert(header.mainHeight - header.trustedHeight  < clientState.trustingPeriod)
+
```
Comment (AntonAndell) [inline docs/adr/ICON-lightclient.md:263]: assert(clientState.latestHeight - header.mainHeight < clientState.trustingPeriod)

File: contracts/cosmwasm-vm/cw-icon-light-client/src/light_client.rs (hunk 1 above)
Comment (AntonAndell) [reply contracts/cosmwasm-vm/cw-icon-light-client/src/light_client.rs:153]: Yeah that should be fine. The generally should not have timeouts longer than that

**Review by redlarva — APPROVED** [review]
looks good to me

**Pull request merged**